SingularityUI/app/components/requestForm/RequestForm.jsx

```diff
@@ -0,0 +1,560 @@
+              <div className="alert alert-info alert-slim" role="alert">
+                <strong>Note:</strong> changes made below will only affect new tasks
+              </div> :
+              undefined}
+            <div className="form-group slavePlacement">
+              <label htmlFor="slavePlacement" className="control-label">
+                Slave Placement
+              </label>
+              <DropDown
+                id = 'slavePlacement'
+                prop = {{
+                  updateFn: event => {
+                    this.updateField('slavePlacement', event.target.value);
+                  },
+                  forceChooseValue: true,
+                  choices: [
+                    {
+                      value: "",
+                      user: "Default"
+                    },
+                    {
+                      value: "SEPARATE",
+                      user: "Separate"
+                    },
+                    {
+                      value: "OPTIMISTIC",
+                      user: "Optimistic"
+                    },
+                    {
+                      value: "GREEDY",
+                      user: "Greedy"
+                    }
+                  ],
+                  value: this.getValue('slavePlacement') ? this.getValue('slavePlacement') : ""
+                }}
+              />
+            </div>
+            {this.renderRequestTypeSpecificFormFields()}
+            <div id="button-row">
+              <span>
+                <button type="submit" className="btn btn-success btn-lg" disabled={this.cantSubmit() ? 'disabled' : undefined}>
+                  Save
+                </button>
+              </span>
+            </div>
+            {this.props.saveApiCall.error ?
+              <p className='alert alert-danger'>
+                There was a problem saving your request: {this.props.saveApiCall.error.message}
+              </p> :
+              undefined
+            }
+            {this.props.saveApiCall.data && this.props.saveApiCall.data.message ?
+              <p className='alert alert-danger'>
+                There was a problem saving your request: {this.props.saveApiCall.data.message}
+              </p> :
+              undefined
+            }
+          </form>
+        </div>
+      </div>
+    );
+  }
+
+};
+
+function mapStateToProps(state) {
+  return {
+    racks: state.api.racks.data,
+    request: state.api.request ? state.api.request.data : undefined,
+    form: state.form[FORM_ID],
+    saveApiCall: state.api.saveRequest
+  }
+}
+
+function mapDispatchToProps(dispatch) {
+  return {
+    update(formId, fieldId, newValue) {
+      dispatch(modifyField(formId, fieldId, newValue));
+    },
+    clearForm(formId) {
+      dispatch(clearForm(formId));
+    },
+    save(requestBody) {
+      dispatch(SaveAction.trigger(requestBody));
+    }
+  }
+}
+
+export default connect(mapStateToProps, mapDispatchToProps, undefined, {pure: false})(RequestForm);
```

Comment:
This is the simplest solution to the fact that react-redux performs a shallow comparison. props.form is an object, and redux doesn't compare its fields to those of the previous props when deciding whether to tell the page about the new props at all.

An alternative is to have all the fields in form surfaced at the top level. Another is to surface the modifications field. Both of those would be more complex, however, than using `pure: false`.